share/dbtypes_gen.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python
+import os
+import sys
+import json
+
+with open(os.path.join(os.path.dirname(__file__), '..', 'src', 'query_backend.h')) as f:
+    lines = f.readlines()
+
+VERSION = "v"+sys.argv[1]
+
+ALL_BACK = ["SQLite", "HDF5"]
+
+ENUM_START = 0
+
+for i in range(0,len(lines)):
+    if "enum DbTypes {" in lines[i]:
+        ENUM_START = i
```

Comment:
Also add an else clause to the for-loop that raises an error.  That way if we don't find the begining, the whole thing breaks.

ie this should look like:

```python
for i in range(len(lines)):
    if lines[i].strip() == 'enum DbTypes {':
        ENUM_START = i
        break
else:
    raise RuntimeError('could not find the start of the DbTypes enum')
```